mitmproxy/addons/strip_dns_https_records.py

```diff
@@ -0,0 +1,29 @@
+from mitmproxy import ctx
+from mitmproxy import dns
+from mitmproxy.net.dns import types
+
+
+class StripDnsHttpsRecords:
+    def load(self, loader):
+        loader.add_option(
+            "strip_ech",
+            bool,
+            True,
+            "Strip Encrypted ClientHello (ECH) data from DNS HTTPS records so that mitmproxy can generate matching certificates.",
+        )
+
+    def dns_response(self, flow: dns.DNSFlow):
+        assert flow.response
+        if ctx.options.strip_ech:
+            for answer in flow.response.answers:
+                if answer.type == types.HTTPS:
+                    answer.https_ech = None
+        if not ctx.options.http3:
+            for answer in flow.response.answers:
+                if (
+                    answer.type == types.HTTPS
+                    and answer.https_alpn is not None
+                    and b"h3" in answer.https_alpn
```

Comment:
alpn values can also be something like `h3-29` or `h3-28` that'd represent http3 as well